rundetection/specifications.py

```diff
@@ -47,9 +46,20 @@ def _load_rules_from_api(self) -> None:
         spec: dict[str, Any] = response.json()
         logger.info("Response from API for spec is: \n%s", spec)
         self._rules = [rule_factory(key, value) for key, value in spec.items()]
+        self._order_rules()
         self.loaded_time = datetime.datetime.now(tz=datetime.UTC)
         logger.info("Loaded instrument specification for: %s at: %s", self._instrument, self.loaded_time)
 
+    def _order_rules(self):
+        """
+        Sometimes we need to ensure some rules end up at the end of the list, notably those with stitch in the name
+        """
+        for rule in self._rules:
+            # We need to ensure rules that do a stitch, or any that added extra jobs, need to come last.
+            if rule.should_be_last:
```

Comment:
This does one pass over the rules. What if the group of rules that get pushed to the end still need to reordered amonst themselves? 

With a list of 1-10, assuming rule 1, 3 and 9 get pushed to the end of list. What if the result should be [remaining old list items, 3, 1, 9]?

Reply:
Until this is relevant I happy for it to be semi-undefined, it is defined it's just unknown. We will likely only have 1 rule that needs to be last, there is reasoning as to why it needs to be last as it copies other rule outputs to new "extra jobs".